src/torchmetrics/functional/image/d_s.py

```diff
@@ -0,0 +1,248 @@
+
+        p: Order of the norm applied on the difference.
+        ws: Window size of the filter applied to degrade the high resolution panchromatic image.
+        reduction: A method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+    Return:
+        Tensor with SpatialDistortionIndex score
+
+    Raises:
+        ValueError:
+            If ``preds`` and ``pan`` don't have the same dimension.
+        ValueError:
+            If ``ms`` and ``pan_lr`` don't have the same dimension.
+        ValueError:
+            If ``preds`` and ``pan`` don't have dimension which is multiple of that of ``ms``.
+
+    Example:
+        >>> _ = torch.manual_seed(42)
+        >>> preds = torch.rand([16, 3, 32, 32])
+        >>> target = {
+        ...     'ms': torch.rand([16, 3, 16, 16]),
+        ...     'pan': torch.rand([16, 3, 32, 32]),
+        ... }
+        >>> preds, target = _spatial_distortion_index_update(preds, target)
+        >>> _spatial_distortion_index_compute(preds, target)
+        tensor(0.0090)
+
+    """
+    length = preds.shape[1]
+
+    ms = target["ms"]
+    pan = target["pan"]
+    pan_lr = target["pan_lr"] if "pan_lr" in target else None
+
+    preds_h, preds_w = preds.shape[-2:]
+    ms_h, ms_w = ms.shape[-2:]
+    pan_h, pan_w = pan.shape[-2:]
+    if preds_h != pan_h:
+        raise ValueError(f"Expected `preds` and `pan` to have the same height. Got preds: {preds_h} and pan: {pan_h}")
+    if preds_w != pan_w:
+        raise ValueError(f"Expected `preds` and `pan` to have the same width. Got preds: {preds_w} and pan: {pan_w}")
+    if preds_h % ms_h != 0:
+        raise ValueError(
+            f"Expected height of `preds` to be multiple of height of `ms`. Got preds: {preds_h} and ms: {ms_h}."
+        )
+    if preds_w % ms_w != 0:
+        raise ValueError(
+            f"Expected width of `preds` to be multiple of width of `ms`. Got preds: {preds_w} and ms: {ms_w}."
+        )
+    if pan_h % ms_h != 0:
+        raise ValueError(
+            f"Expected height of `pan` to be multiple of height of `ms`. Got preds: {pan_h} and ms: {ms_h}."
+        )
+    if pan_w % ms_w != 0:
+        raise ValueError(f"Expected width of `pan` to be multiple of width of `ms`. Got preds: {pan_w} and ms: {ms_w}.")
+    if ws >= ms_h or ws >= ms_w:
+        raise ValueError(f"Expected `ws` to be smaller than dimension of `ms`. Got ws: {ws}.")
+
+    if pan_lr is not None:
+        pan_lr_h, pan_lr_w = pan_lr.shape[-2:]
+        if pan_lr_h != ms_h:
+            raise ValueError(
+                f"Expected `ms` and `pan_lr` to have the same height. Got ms: {ms_h} and pan_lr: {pan_lr_h}."
+            )
+        if pan_lr_w != ms_w:
+            raise ValueError(
+                f"Expected `ms` and `pan_lr` to have the same width. Got ms: {ms_w} and pan_lr: {pan_lr_w}."
+            )
```

Comment:
all this is just input checking so it should probably just be moved to the `_update` method

Reply:
thanks for the suggestion. i've moved the checking of tensor to `_update` method